File: 370. Range Addition/solution.py
class Solution:
    def getModifiedArray(self, length: int, updates: List[List[int]]) -> List[int]:
        state = [0]*length
        ops = [0]*(length+1)
        
        # gather operations
        for x, y, z in updates:
            ops[x] += z
            ops[y+1] -= z
                
        # perform running sum
        total = 0
        for i in range(length):
            total += ops[i]
            state[i] = total
            
        return state

File: 370. Range Addition/solution.java
class Solution {
    public int[] getModifiedArray(int n, int[][] updates) {
        int[] corners = new int[n+1];
        int[] array = new int[n];
        for (int i = 0; i < n; i++) {
            corners[i] = 0;
            array[i] = 0;
        }
        for (int[] update: updates) {
            corners[update[0]] += update[2];
            corners[update[1]+1] -= update[2];
        }
        
        // running sum
        int sum = 0;
        for (int i = 0; i < n; i++) {
            sum += corners[i];
            array[i] = sum;
        }
        
        return array;
    }
}
